Survival-DEV/blog-backend
# Blog 37: Refactor the code | PART I

### :dart:  What is been done in this PR:
1. [x] Return the data only needed from the queries instead of the current behavior of `select *` for users service
2. [x] Validate the whole Auth process
3. [x] Update the types
4. [x] Move the frequently called functions into the `helpers` folder 
5. [x] Move swagger out of `helpers` folder
7. [x] Add Redis as a cacheing manager
8. [x] Update the imports paths
9. [x] Handle the errors properly  
10. [x] Renaming
11. [x] Bypass the tests for now
 **Related to #38**

----
### :compass:   How to test it
Steps to test my PR:
1. `npm run ci`
2. make sure to setup your `env` file to have these params:
   <details>
      <summary>Expand to see the params</summary>
    
      > DATABASE_URL=<your db url#>
      > NODE_ENV=DEBUG  //for now
      > logDB=["query", "error"]
      > PORT=<your port#>
      > TRUSTED_IP=["127.0.0.1"]
      > WHITELIST=['http://localhost']
      > ALLOWED_METHODS='GET,HEAD,PUT,PATCH,POST,DELETE,OPTIONS'
      > JWT_VERIFICATION_TOKEN_SECRET='secreto2'
      > JWT_VERIFICATION_TOKEN_EXPIRATION_TIME='60s'
      > EMAIL_CONFIRMATION_URL='http://localhost:5000/api/confirm-email/'
      > SEND_GRID_ACCESS_KEY=`<your sendgird key>`
      > FROM_EMAIL=`your sendgrid email`
      > SENDGRID_TEMPLATE_ID=d-3285e7571ee64e8fafedb163996e5a55
      >CONTACT_EMAIL= `<some_mail for_swagger_configs>`
   </details>
4. `docker-compose up`   --to fire redis 
3. `npm run start:dev`   --in another terminal
